data/scenarios/Testing/380-counting-objective.yaml

```diff
@@ -0,0 +1,36 @@
+version: 1
+name: Evaluate count of an item
+description: |
+  The goal criteria is exactly the same as
+  379-possession-objective but evaluated
+  with the "count" function instead.
+creative: false
+objectives:
+  - goal:
+      - |
+        Get 1 tree.
+    condition: |-
+      as base {
+        itemCount <- count "tree";
+        return $ itemCount > 0;
```

Comment:
This does not succeed even after `grab`-bing the `tree`.

Reply:
I've been poking at this a bit.  Curiously, I've found that it does work correctly if the test is wrapped in a `try` block, like this:
```
      as base {
        try {
          itemCount <- count "tree";
          return $ itemCount > 0
        } { return false };
      };
```
In theory this should not make a difference.

Reply:
The runtime code for `has` and `count` is almost identical.  The one big difference between them which seems like it might be relevant is that `has` does not require any capabilities to use, whereas `count` requires a `counter`.  But in theory the condition checking robot should be able to use `count` whether `base` has a `counter` or not (and indeed, if I add a `counter` installed on `base` it does not change anything).

Reply:
Hmm, maybe base can not count or something?

Reply:
I really want to figure out what exceptions are being thrown but I'm not sure how to export that information. @xsebek, any ideas?

Reply:
@byorgey I just checked by printing exceptions from the hypothetical check and _there are none_.

I just added this:
```Haskell
sendIO $ putStr "ERROR: "
sendIO $ print m
```
here:
https://github.com/swarm-game/swarm/blob/e73a2849e39bee48a8875bc53fba1a654285d074/src/Swarm/Game/Step.hs#L137-L146

Weirdly this is also a transformation that fixes the condition:
```
t <- time;
if (t < 0) {return true} {
    as base {
        itemCount <- count "tree";
        return $ itemCount > 0;
    };
}
```

Reply:
Looking at debug logs (see the [debug branch](https://github.com/swarm-game/swarm/tree/debug-count)), the count is correct:
![image](https://user-images.githubusercontent.com/44544735/201556049-3388092d-a6de-47c7-a999-efaa550f350c.png)